lisp/ess-inf.el

```diff
@@ -250,18 +241,18 @@ This may be useful for debugging."
      ((and (not (comint-check-proc (current-buffer)))
            (derived-mode-p 'inferior-ess-mode))
       (current-buffer))
-     ;; Take the *R:N* buffer if already exists (and contains dead proc!)
-     ;; fixme: buffer name might have been changed, iterate over all
-     ;; inferior-ess buffers.
-     ((get-buffer inf-name))
      ;; Pick up a transcript file
      (ess-ask-about-transfile
       (let ((transfilename (read-file-name
                             "Use transcript file (default none):" nil "")))
         (if (string= transfilename "")
             (get-buffer-create inf-name)
-          (find-file-noselect (expand-file-name transfilename)))))
-     ;; Create a new buffer
+          (let ((buf (find-file-noselect (expand-file-name transfilename))))
+            (if (get-buffer-process buf)
+                (error "Can't start a transcripted session if it already exists")
```

Comment:
I don't think we should blindly throw errors here. I think if the dead-proc buffer exists, even if ess-ask-about-transfile is t, we should not load new transfile, just like the old logic did.

Reply:
This error is when there is a live proc, not a dead proc. What do you mean by "not load new transfile"? The old logic just exited from creating a new process. This behaviour is no different and makes control flow simpler.

Reply:
The old logic would just return a file with "inf-name" if it exists, live process or not. This is changed now in case `ess-ask-about-transfile` is non-nil. Throwing if `ess-ask-about-transfile` is non-nil doesn't seem right. At the very least one should throw regardless of that variable.

Reply:
If there were a live process, this if branch would exit: https://github.com/emacs-ess/ESS/blob/661daee75deace95ccecb37014bde6b475004e31/lisp/ess-inf.el#L194

Hmm I think I'm missing a `comint-check-proc` actually. Is this what you mean?

Reply:
>  if branch would exit:

It doesn't matter what the consumer of this function does. If a logic has been extracted into a function it should be self-contained and behave as documented. 

> comint-check-proc actually.

I don't have a good picture of what the new code does, but I think it's not only that. One needs to ensure that the name returned by that func-gen doesn't name an existing buffer, right?

Reply:
>  If a logic has been extracted into a function it should be self-contained and behave as documented.

Agreed, though that's an internal function so we're allowed to get there gradually. I'll try to find a solution.

> One needs to ensure that the name returned by that func-gen doesn't name an existing buffer, right?

Yes I think so, cf other thread. I think I know how to fix this.